src/plugins/simulator/model/simulator-content.ts

```diff
@@ -111,20 +111,22 @@ export const SimulatorContentModel = TileContentModel
     updateAfterSharedModelChanges() {
       // nothing to do here
     },
-    step() {
+    step(readOnly?: boolean) {
```

Comment:
Is this what you were making reference to today - in terms of building in context awareness within the "program" execution?

Reply:
Yes. The `step` function is greatly reduced in `readOnly` mode in this PR, which prevents the My Work copy of the document from modifying the shared variables at the same time the working document, but also prevents documents that are only in `readOnly` mode (like published documents) from ever updating.